File: InventoryApp.Data/Entities/Item.cs
namespace InventoryApp.Data.Entities;

/// <summary>
/// An object that will be tracked in the inventory database.
/// </summary>
public sealed class Item
{
    /// <summary>
    /// Primary key ID.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// The name of the item.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// The description of the item.
    /// </summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// The quantity of the item that is in stock.
    /// </summary>
    public long Quantity { get; set; }
}

File: InventoryApp.Client/src/app/models/item.ts
/**
 * Model for an item object.
 */
export interface Item {

  /**
   * Primary key id (GUID).
   */
  id: string;

  /**
   * The name of the item.
   */
  name: string;

  /**
   * The description of the item.
   */
  description: string;

  /**
   * The amount of the item currently available.
   */
  quantity: number;
}
